packages/Tag/src/Tag.js

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Close from "@paprika/icon/lib/Times";
+import * as sc from "./Tag.styles";
+import * as types from "./types";
+
+export function Delete({ isDisabled, size, onRemove = () => {} }) {
+  return (
+    <sc.Delete isDisabled={isDisabled} size={size} data-pka-anchor="tag.delete" onClick={onRemove}>
+      <Close />
+    </sc.Delete>
+  );
+}
+
+Delete.propTypes = {
+  onRemove: PropTypes.func.isRequired,
+  size: PropTypes.oneOf([types.MEDIUM, types.LARGE]).isRequired,
+  isDisabled: PropTypes.bool,
+};
+
+Delete.defaultProps = {
+  isDisabled: false,
+};
+
+export default function Tag(props) {
+  const { children, onRemove, size, onClick, a11yText, isDisabled, ...moreProps } = props;
+  const isChildString = typeof children === "string";
+
+  const handleRemove = e => {
+    onRemove();
+    e.stopPropagation();
+  };
+
+  const content = (
+    <>
+      <sc.Ellipsis isChildString={isChildString} size={size}>
+        {children}
+      </sc.Ellipsis>
+      {onRemove ? <Delete isDisabled={isDisabled} onRemove={handleRemove} size={size} /> : null}
+    </>
+  );
+
+  if (onClick) {
+    if (!a11yText) {
+      console.warn("Tag component should include an a11yText prop if an onClick is provided.");
+    }
+    return (
+      <sc.RawButtonTag a11yText={a11yText} isDisabled={isDisabled} onClick={onClick} {...moreProps}>
+        {content}
+      </sc.RawButtonTag>
+    );
+  }
+
+  return (
+    <sc.Tag data-pka-anchor="tag" {...moreProps}>
+      {content}
+    </sc.Tag>
+  );
+}
+
+Tag.types = {
+  size: PropTypes.oneOf([types.MEDIUM, types.LARGE]),
```

Comment:
This shouldn't be a `PropTypes` definition, just an object of valid sizes.  Probably best to define in `./types.js` then just follow the same pattern as below (`size: types.sizes,`)